momento-sdk/src/main/java/momento/sdk/ISubscribeCallOptions.java

```diff
@@ -0,0 +1,23 @@
+package momento.sdk;
+
+import momento.sdk.responses.topic.TopicMessage;
+
+/** Represents options for a topic subscription callback. */
+public interface ISubscribeCallOptions {
+  /**
+   * Called when a new message is received on the subscribed topic.
+   *
+   * @param message The received topic message.
+   */
+  void onItem(TopicMessage message);
```

Comment:
you are doing callback-style here. In some SDKs we do callback-style, and in some we do iterator style with a blocking iterator method like `nextMessage()`. I believe golang and .NET do that iterator style.

This usually boils down to how the gRPC interface for streams is presented to us by the grpc lib. I assume in Java it's callback-style?

@nand4011 I believe that you were the one that did the iterator-style implementation for .NET, right? can you weigh in on whether that pattern would be possible/desirable in Java, based on what you've seen of the grpc lib's support for streaming?

I'm not opposed to callback-style, but just want to make sure we have investigated the options and made a deliberate, informed choice.

Reply:
It looks like we can supply a `ClientCallStreamObserver` to the request that lets you disable automatic message requesting and provides a request method so you can force it into an iterator style pattern.

Reply:
@nand4011 cool, thanks. since you are among our most experienced Java engineers and you also worked on the iterator-style pattern in the .NET SDK, do you have thoughts on whether it's worth it?

Also do we know what thread the callbacks get called on if we stick with the callback pattern?